### PR TITLE
HA flows Part 12: Add H&S partial update operation

### PR DESCRIPTION
**Pull request by niksv**
Part of #5061

### CODE REVIEW

**Review by artHalMirantis — APPROVED** [review]
from the testing prospective  - its ready to merge
the tests are enough to cover basic needs/cases

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateService.java [hunk 1]
```diff
@@ -96,6 +136,230 @@ public void handleUpdateRequest(
...
+                || isFieldUpdated(request.getEncapsulationType(), haFlow.getEncapsulationType())
+                || isUpdateRequiredByDiverseFlowIdField(request, haFlow);
+    }
+
+    private static boolean isUpdateRequiredByPathComputationStrategy(
+            HaFlowPartialUpdateRequest request, HaFlow haFlow) {
+        PathComputationStrategy strategy = Optional.ofNullable(request.getPathComputationStrategy())
+                .orElse(haFlow.getPathComputationStrategy());
+        boolean strategyIsLatencyBased = LATENCY_BASED_STRATEGIES.contains(strategy);
+        boolean changedMaxLatency = isFieldUpdated(request.getMaxLatency(), haFlow.getMaxLatency());
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateService.java:293]: minor, non-blocking: the value on 293 is only needed if 292 is true and 295 is false, but it is calculated every time even when we know that the result will be discarded. I'd suggest to split/extract it in separate methods and call them only when needed (for example, as an operand in `||`)

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+                            .endpoint(new FlowPartialUpdateEndpoint(SWITCH_ID_3, PORT_3, VLAN_3, INNER_VLAN_3))
+                            .description(DESCRIPTION_2)
+                            .build()))
+            .diverseFlowId(HA_FLOW_ID_2)
+            .build();
+
+
+    @Mock
+    private FlowGenericCarrier carrier;
+    HaFlowUpdateService service;
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java:94]: could be private. I'd separate fields in different blocks: all mocks, all expected results, and so on.

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+
+    @Test
+    public void isFullUpdateRequiredHaPathId() {
+        assertFalse(service.isFullUpdateRequired(patchBuilder().haFlowId(HA_FLOW_ID_1).build(), HA_FLOW));
+        assertFalse(service.isFullUpdateRequired(patchBuilder().haFlowId(HA_FLOW_ID_2).build(), HA_FLOW));
+    }
+
+    @Test
+    public void isFullUpdateRequiredSharedSwitch() {
+        assertTrue(service.isFullUpdateRequired(patchBuilder().sharedEndpoint(
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java:116]: the expected and actual are swapped here? Actually not super clear which is the expected. Variable names could really help here.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateService.java (hunk 1 above)
Comment (niksv) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateService.java:293]: good point

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java (hunk 2 above)
Comment (niksv) [reply src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateServiceTest.java:94]: will fix in next PR

**Pull request merged**